### ISSUE TITLE
test_selection_listener_disconnected fails under Python 3.8

### ISSUE THREAD

**Issue by mdickinson** [enthought/traitsui#722]
I'm seeing a test failure on Python 3.8 with the latest release of TraitsUI, when running tests with `nosetests traitsui`. I don't see this failure with `unittest`, and I haven't tracked down the rogue `raise StopIteration`.
```
======================================================================
ERROR: Check that selection listeners get correctly disconnected
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/mdickinson/.venvs/traits-3.8/lib/python3.8/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/Users/mdickinson/.venvs/traits-3.8/lib/python3.8/site-packages/traitsui/tests/_tools.py", line 90, in wrapper
    return item(*args, **kwargs)
  File "/Users/mdickinson/.venvs/traits-3.8/lib/python3.8/site-packages/traitsui/tests/editors/test_liststr_editor_selection.py", line 167, in test_selection_listener_disconnected
    list_view.selectionModel().select(mi, QItemSelectionModel.ClearAndSelect)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/contextlib.py", line 120, in __exit__
    next(self.gen)
RuntimeError: generator raised StopIteration

**Comment by mdickinson** [comment]
It's beginning to look as though this is coming from somewhere inside PySide2; I don't see the `StopIteration` when I replace PySide2 with PyQt5.

**Comment by rahulporuri** [comment]
closing as i am unable to reproduce this at the moment - with the following packages and python 3.8.9

```
❯ python -m pip list
Package             Version      Location
------------------- ------------ -----------------------------------------
importlib-metadata  4.0.1
importlib-resources 5.1.2
pip                 21.1
pyface              7.3.0
PySide2             5.15.2
setuptools          56.0.0
shiboken2           5.15.2
traits              6.2.0
traitsui            7.2.0.dev497 c:\users\rporuri\work\github\ets\traitsui
wheel               0.36.2
zipp                3.4.1
```

**Comment by mdickinson** [comment]
@rahulporuri It looks as though this is the relevant PySide bug, which is indeed fixed in PySide 5.15.1: https://bugreports.qt.io/browse/PYSIDE-1313

(See also https://bugs.python.org/issue40789.)

Confirmed that this looks fixed from my end: I no longer see the issue locally with Python 3.8 and the latest Pyface, TraitsUI and PySide (but if I install PySide 5.14.0, I do still see the `StopIteration`).
